### PR TITLE
KTOR-7596 Make multipart Content-Type check case-insensitive

### PR DESCRIPTION
**Pull request by osipxd**
**Subsystem**
Server, `ktor-http-cio`

**Motivation**
[KTOR-7596](https://youtrack.jetbrains.com/issue/KTOR-7596) receiveMultipart fails with "IOException: Failed to parse multipart" when content-type is capitalized

**Solution**
Use case-insensitive check for multipart content-type

### CODE REVIEW

File: ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/JettyKtorHandler.kt [hunk 1]
```diff
@@ -72,7 +71,7 @@ internal class JettyKtorHandler(
     ) {
         try {
             val contentType = request.contentType
-            if (contentType != null && contentType.startsWith("multipart/")) {
+            if (contentType != null && contentType.startsWith("multipart/", ignoreCase = true)) {
```
Comment (e5l) [inline ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/JettyKtorHandler.kt:74]: I would extract constant here

**Review by e5l — APPROVED** [review]
lgtm

File: ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/JettyKtorHandler.kt (hunk 1 above)
Comment (osipxd) [reply ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/JettyKtorHandler.kt:74]: I could add a constant `ContentType.Application.TYPE`. I also thought about adding an operator function `contains`:

```kotlin
public object ContentType {
    public object Application {
        // ...

        public operator fun contains(contentType: CharSequence): Boolean =
            contentType.startsWith("$APPLICATION/", ignoreCase = true)
    }
}
```
And then check the content type using `in`: 
```kotlin
if (contentType in ContentType.Application) { ... }
```
What do you think?

Comment (osipxd) [reply ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/JettyKtorHandler.kt:74]: I'll make it in a separate PR

Comment (e5l) [reply ktor-server/ktor-server-jetty/jvm/src/io/ktor/server/jetty/JettyKtorHandler.kt:74]: @bjhham, what do you think?

**Pull request merged**